File: backup/be/pharmacy 8.7.2022 1500/pharmacy/src/main/java/com/c1221g1/pharmacy/service/employee/impl/EmployeeService.java
package com.c1221g1.pharmacy.service.employee.impl;

import com.c1221g1.pharmacy.entity.employee.Employee;
import com.c1221g1.pharmacy.entity.user.UserRole;

import com.c1221g1.pharmacy.repository.employee.IEmployeeRepository;
import com.c1221g1.pharmacy.service.employee.IEmployeeService;
import org.springframework.beans.factory.annotation.Autowired;
import com.c1221g1.pharmacy.entity.employee.Employee;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class EmployeeService implements IEmployeeService {

    @Autowired
    private IEmployeeRepository iEmployeeRepository;

    /**
     * Created by: TrungTVH
     * Date created: 5/7/2022
     * function: get medicine list dùng tạm
     */
    @Override
    public List<Employee> getList() {
        return this.iEmployeeRepository.findAll();
    }

    /**
     * this function use to get all page Employee
     *
     * @author GiangTB
     * @Time 15:30 29/06/2022
     */
    public Page<Employee> getAllEmployee(String employeeIdVal, String employeeNameVal,
                                         String positionVal, String employeeAddressVal, String employeePhoneVal, Pageable pageable) {
        return this.iEmployeeRepository.getAllEmployee(
                "%" + employeeIdVal + "%",
                "%" + employeeNameVal + "%",
                "%" + positionVal + "%",
                "%" + employeeAddressVal + "%",
                "%" + employeePhoneVal + "%", pageable);
    }

    /**
     * this function use to find Employee by id
     *
     * @author GiangTB
     * @Time 17:00 29/06/2022
     */
    @Override
    public Employee findEmployeeById(String id) {
        return this.iEmployeeRepository.findEmployeeById(id);
    }

    /**
     * this function use to delete employee from page (not delete database)
     *
     * @author GiangTB
     * @Time 17:00 29/06/2022
     */
    @Override
    public void deleteEmployeeById(String id) {
        this.iEmployeeRepository.deleteEmployeeById(id);
    }


    /**
     * this function use to get all list Employee
     *
     * @author GiangTB
     * @Time 16:00 29/06/2022
     */
    @Override
    public List<Employee> getListEmployee() {
        return this.iEmployeeRepository.getListEmployee();
    }
/*
  Created by TamNA
  Time: 13:35:00 01/07/2022
  Function: Save Emoloyee
*/

    @Override
    public void saveEmployee(Employee employee) {
        this.iEmployeeRepository.save(employee);
    }
/*
  Created by TamNA
  Time: 13:35:00 01/07/2022
  Function:  Update Emoloyee
*/

    @Override
    public void updateEmployee(Employee employee) {
        this.iEmployeeRepository.updateEmployee(employee.getEmployeeName(), employee.getEmployeeImage(), employee.getEmployeeAddress(), employee.getEmployeePhone(), employee.getEmployeeDateStart(), employee.getEmployeeNote(), employee.getPosition(), employee.getEmployeeUsername(), employee.getEmployeeId());
    }
}

File: backup/fe/sprint1 6.7.2022 2000/sprint1/src/app/service/employee/employee.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {environment} from '../../../environments/environment';
import {Observable} from 'rxjs';
const API_URL = `${environment.apiUrl}`;
import * as empty from 'firebase/empty-import';
import {Employee} from '../../model/employee/employee';
@Injectable({
  providedIn: 'root'
})
export class EmployeeService {
  constructor(private http: HttpClient) { }

  /**
   * this function use to get all page Employee
   *
   * @author GiangTB
   * @Time 15:30 02/07/2022
   */
  public getAllEmployee(request): Observable<Employee[]> {
    const params = request;
    return this.http.get<Employee[]>(API_URL + `/api/manager-employee/employees`, {params});
  }

  /**
   * this function use to get all page Employee
   *
   * @author GiangTB
   * @Time 19:00 02/07/2022
   */
  public deleteEmployeeById(id: string): Observable<Employee> {
    return this.http.delete<Employee>(`${API_URL}/api/manager-employee/employees/${id}`)
  }
  /*
    Created by TamNA
    Time: 08:50:00 03/07/2022
    Function:  save employee
  */
  saveEmployee(employee): Observable<Employee> {
    return this.http.post<Employee>('http://localhost:8080/api/manager-employee/employees/', employee);
  }


  /*
     Created by TamNA
     Time: 08:55:00 03/07/2022
     Function:  save employee
   */
  findEmployeeById(id: string): Observable<Employee> {
    return this.http.get<Employee>(`${'http://localhost:8080/api/manager-employee/employees'}/${id}`);
  }

  /*
     Created by TamNA
     Time: 09:00:00 03/07/2022
     Function:  save employee
   */
  updateEmployee(id: string, employee: Employee): Observable<Employee> {
    return this.http.patch<Employee>(`${'http://localhost:8080/api/manager-employee/employees'}/${id}`, employee);
  }
}
